src/main/java/com/stuypulse/robot/subsystems/Drivetrain.java

```diff
@@ -29,21 +28,44 @@
 import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
 import edu.wpi.first.wpilibj2.command.SubsystemBase;
 
-public class Drivetrain extends SubsystemBase {
+import com.stuypulse.robot.util.Encoder;
 
-    // Enum used to store the state of the gear
-    public static enum Gear {
-        HIGH,
-        LOW
-    };
+public class Drivetrain extends SubsystemBase {
 
+    public enum Gear {
+
+        // Create all the gear settings with their ratios
+    
+        HIGH(DrivetrainSettings.Encoders.HIGH_GEAR_DISTANCE_PER_ROTATION),
+        LOW(DrivetrainSettings.Encoders.LOW_GEAR_DISTANCE_PER_ROTATION);
+    
+        // Store the gear ratio
+    
+        private final Number ratio;
```

Comment:
this isnt a comment, but I feel guilty about how badly we abuse this number class.